Goal.gd

```diff
@@ -0,0 +1,17 @@
+extends Node
+
+class_name Goal
+
+var key_sequence: Array
+
+func _init(initial_key_sequence):
+	key_sequence = initial_key_sequence
```

Comment:
Just a data class to hold the target key presses for each beat. This is an array, so the idea is one goal maps to one measure. Down to change if that's not intuitive.